src/Workspaces/Core/Portable/Workspace/Solution/Solution.CompilationTracker.cs

```diff
@@ -794,6 +788,58 @@ public IEnumerable<SyntaxTree> GetSyntaxTreesWithNameFromDeclarationOnlyCompilat
                 return clone.GetSymbolsWithName(predicate, filter, cancellationToken).SelectMany(s => s.DeclaringSyntaxReferences.Select(r => r.SyntaxTree));
             }
 
+            public Task<bool> HasCompleteReferencesAsync(Solution solution, CancellationToken cancellationToken)
+            {
+                var state = this.ReadState();
+
+                if (state.HasCompleteReferences.HasValue)
+                {
+                    return Task.FromResult(state.HasCompleteReferences.Value);
+                }
+                else
+                {
+                    return GetOrBuildCompilationInfoAsync(solution, lockGate: true, cancellationToken: cancellationToken)
+                        .ContinueWith(t => t.Result.HasCompleteReferences, cancellationToken, TaskContinuationOptions.ExecuteSynchronously | TaskContinuationOptions.OnlyOnRanToCompletion, TaskScheduler.Default);
```

Comment:
done.

it looks like for cross p2p reference, if skeleton assembly is not required to update (no top level change), we used to not move compilation to current. this change will force it to be. but I think that probably will be fine.